Getting error while using Node v4.2.4

Hello~

Could you help me diagnose my error? Everything was working fine when I was using Node v 0.12.7.

```
return process.dlopen(module, path._makeLong(filename));
                 ^

Error: dlopen(/Users/angie/projects/node-projects/search-api/node_modules/node-libcurl/lib/binding/node_libcurl.node, 1): Library not loaded: /Users/travis/lib/libcurl.4.dylib
  Referenced from: /Users/angie/projects/node-projects/search-api/node_modules/node-libcurl/lib/binding/node_libcurl.node
  Reason: Incompatible library version: node_libcurl.node requires version 9.0.0 or later, but libcurl.4.dylib provides version 7.0.0
```

OS: OSX 10.10.5
xcode: 7.2
libcurl: 7.43.0

Thanks!

I didn't have homebrew on the computer I'm working in so i installed it and now i have libcurl v7.46.0

running `node .` i get a different error: 

```
module.js:460
  return process.dlopen(module, path._makeLong(filename));
                 ^

Error: Module version mismatch. Expected 46, got 14.
    at Error (native)
    at Object.Module._extensions..node (module.js:460:18)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:313:12)
    at Module.require (module.js:366:17)
    at require (module.js:385:17)
    at Object.<anonymous> (/Users/angie/projects/node/search-api/node_modules/node-libcurl/lib/Curl.js:27:14)
    at Module._compile (module.js:435:26)
    at Object.Module._extensions..js (module.js:442:10)
    at Module.load (module.js:356:32)
```

I also tried running tried running `npm install node-libcurl --build-from-source` and I get this error: 

```
npm WARN enoent ENOENT, open '/Users/angie/projects/node/search-api/node_modules/bin/package.json'
npm WARN enoent ENOENT, open '/Users/angie/projects/node/search-api/node_modules/etc/package.json'
npm WARN enoent ENOENT, open '/Users/angie/projects/node/search-api/node_modules/lib/package.json'
npm WARN search-api@1.0.0 No license field.
npm ERR! Darwin 15.2.0
npm ERR! argv "node" "/Users/angie/projects/node/search-api/node_modules/.bin/npm" "install" "node-libcurl" "--build-from-source"
npm ERR! node v0.12.7
npm ERR! npm  v3.5.2
npm ERR! path /Users/angie/projects/node/search-api/node_modules/.staging/abbrev-ad080861
npm ERR! code ENOENT
npm ERR! errno -2

npm ERR! enoent ENOENT, rename '/Users/angie/projects/node/search-api/node_modules/.staging/abbrev-ad080861'
npm ERR! enoent ENOENT, rename '/Users/angie/projects/node/search-api/node_modules/.staging/abbrev-ad080861'
npm ERR! enoent This is most likely not a problem with npm itself
npm ERR! enoent and is related to npm not being able to find a file.
npm ERR! enoent 

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/angie/projects/node/search-api/npm-debug.log
```

Thank you so much for helping me!

--- a/.travis.yml
+++ b/.travis.yml
@@ -85,7 +85,7 @@ before_install:
   - cd "$TAG"
   - CURL_PREFIX=$HOME
   - if [[ $TRAVIS_OS_NAME == "osx" ]]; then
-      CURL_PREFIX=/usr/local
+      CURL_PREFIX=/usr/local;
     fi
   - ./buildconf && ./configure --prefix=$CURL_PREFIX --with-ssl && make && make install
   - cd $TRAVIS_BUILD_DIR

--- a/.travis.yml
+++ b/.travis.yml
@@ -83,7 +83,7 @@ before_install:
       git clone --branch "$TAG" --depth 1 https://github.com/bagder/curl.git "$TAG";
     fi
   - cd "$TAG"
-  - ./buildconf && ./configure --prefix=$HOME --with-ssl && make && make install
+  - ./buildconf && ./configure --prefix=/opt/local --with-ssl && make && make install
   - cd $TRAVIS_BUILD_DIR
   # output curl version info
   - curl --version

--- a/.travis.yml
+++ b/.travis.yml
@@ -83,7 +83,11 @@ before_install:
       git clone --branch "$TAG" --depth 1 https://github.com/bagder/curl.git "$TAG";
     fi
   - cd "$TAG"
-  - ./buildconf && ./configure --prefix=/usr/local --with-ssl && make && make install
+  - CURL_PREFIX=$HOME
+  - if [[ $TRAVIS_OS_NAME == "osx" ]]; then
+      CURL_PREFIX=/usr/local
+    fi
+  - ./buildconf && ./configure --prefix=$CURL_PREFIX --with-ssl && make && make install
   - cd $TRAVIS_BUILD_DIR
   # output curl version info
   - curl --version

--- a/.travis.yml
+++ b/.travis.yml
@@ -83,7 +83,7 @@ before_install:
       git clone --branch "$TAG" --depth 1 https://github.com/bagder/curl.git "$TAG";
     fi
   - cd "$TAG"
-  - ./buildconf && ./configure --prefix=/opt/local --with-ssl && make && make install
+  - ./buildconf && ./configure --prefix=/usr/local --with-ssl && make && make install
   - cd $TRAVIS_BUILD_DIR
   # output curl version info
   - curl --version

--- a/binding.gyp
+++ b/binding.gyp
@@ -88,6 +88,7 @@
                         ],
                         'LD_RUNPATH_SEARCH_PATHS': [
                           '/opt/local/lib',
+                          '/usr/local/opt/curl/lib',
                           '/usr/local/lib',
                           '/usr/lib'
                         ]

--- a/binding.gyp
+++ b/binding.gyp
@@ -88,6 +88,7 @@
                         ],
                         'LD_RUNPATH_SEARCH_PATHS': [
                           '/opt/local/lib',
+                          '/usr/local/lib',
                           '/usr/lib'
                         ]
                     },

--- a/binding.gyp
+++ b/binding.gyp
@@ -85,13 +85,17 @@
                         'WARNING_CFLAGS':[
                             '-Wno-c++11-narrowing',
                             '-Wno-constant-conversion'
+                        ],
+                        'LD_RUNPATH_SEARCH_PATHS': [
+                          '/opt/local/lib',
+                          '/usr/lib'
                         ]
                     },
                     'include_dirs' : [
                         '<!@(node "<(module_root_dir)/tools/curl-config.js" --cflags | sed s/-I//g)'
                     ],
                     'libraries': [
-                        '<!@(node "<(module_root_dir)/tools/curl-config.js" --libs)'
+                        '-L <!@(node "<(module_root_dir)/tools/curl-config.js" --prefix)/lib -lcurl'
                     ]
                 }]
             ]
@@ -105,7 +109,21 @@
                     'files': [ '<(PRODUCT_DIR)/<(module_name).node' ],
                     'destination': '<(module_path)'
                 }
-            ]
+            ],
+            'conditions': [['OS == "mac"', {
+                'postbuilds': [
+                    {
+                      'postbuild_name': '@rpath for libcurl',
+                      'action': [
+                        'install_name_tool',
+                        '-change',
+                        '<!@(otool -D `curl-config --prefix`/lib/libcurl.dylib | sed -n 2p)',
+                        '@rpath/libcurl.dylib',
+                        '<(module_path)/<(module_name).node'
+                      ],
+                    },
+                ]
+            }]]
         }
     ]
 }

